domains/certificates/Query.API/API.IntegrationTests/Shared/Migrations/UpdateTechnologyCodesMigrationTests.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Threading.Tasks;
+using API.ContractService;
+using API.Data;
+using API.IntegrationTests.Testcontainers;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Npgsql;
+using Xunit;
+
+namespace API.IntegrationTests.Shared.Migrations;
+
+public class UpdateTechnologyCodesMigrationTests : IAsyncDisposable
+{
+    private readonly PostgresContainer container = new();
+
+    [Fact]
+    public async Task ApplyMigration_WhenContractsHaveEmptyTechnologyCodes_UpdatesFieldsCorrectly()
+    {
+        await using var dbContext = await CreateNewCleanDatabase();
+
+        var migrator = dbContext.GetService<IMigrator>();
+        await migrator.MigrateAsync("20231107095405_AddTechnologyToCertificateIssuingContract");
+
+        var contractId = Guid.NewGuid();
+        await InsertContractWithEmptyTechnologyFields(dbContext, contractId, "123456789", "DK1");
+
+        var contractBeforeMigration = await dbContext.Contracts.FindAsync(contractId);
+        contractBeforeMigration!.Technology.FuelCode.Should().Be("");
+        contractBeforeMigration.Technology.TechCode.Should().Be("");
+
+        var applyMigration = () => migrator.MigrateAsync("20231114121811_UpdateEmptyTechnologyCodes");
+        await applyMigration.Should().NotThrowAsync();
+
+        await using var newDbContext = await CreateNewCleanDatabase();
```

Comment:
This method does not create a new clean database, so this is really hard to read/understand. Also, I am not sure why you want a new dbContext here...